### PR TITLE
Collection walk method bug fix when specific callback function

### PR DESCRIPTION
**Pull request by jalogut**
When a specific class is set as callback on a collection->walk, it works only for the first item in the collection. The problem is that array_unshift modifies directly the value of $args adding the $item object at the beginning. That works well for the first iteration but after that $args grow and grow keeping previous $item objects from previous iterations.

How to test:

Just try an a collection->walk with more than 1 item. Only the callback for the first element is successful.

``` php
$this->objectCollection
    ->addFieldToFilter('id', array('in' => array(1,2,3)))
    ->walk(array($this, 'saveCollection'), array('value'));
```

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=5742) <br/>All committers have signed the CLA.

File: lib/internal/Magento/Framework/Data/Collection.php [hunk 1]
```diff
@@ -500,16 +500,19 @@ public function clear()
...
                 $cb = [$item, $callback];
             } else {
                 $cb = $callback;
-                array_unshift($args, $item);
+                array_unshift($params, $item);
             }
-            $results[$id] = call_user_func_array($cb, $args);
+            //@codingStandardsIgnoreStart
+            $results[$id] = call_user_func_array($cb, $params);
+            //@codingStandardsIgnoreEnd
```
Comment (orlangur) [inline lib/internal/Magento/Framework/Data/Collection.php]: What needs to be ignored here?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Data/Collection.php (hunk 1 above)
Comment (jalogut) [reply lib/internal/Magento/Framework/Data/Collection.php]: Nothing, I was just getting a warning that you should not use "call_user_func_array". I removed these coding standard ignores

**Comment by magento-team** [comment]
@jalogut Thank you for the contribution

**Pull request merged**